README.md

```diff
@@ -36,9 +37,82 @@ You can install _Django JSON Agg_ via [pip] from [PyPI]:
 $ pip install django-json-agg
 ```
 
+And add it to Django settings `INSTALLED_APPS`:
+
+```python
+INSTALLED_APPS = [
+    # other apps
+    'json_agg',
+]
+```
+
 ## Usage
 
-Please see the [Command-line Reference] for details.
+Let's assume you have the following models
+
+```python
+class Post(models.Model):
+    """Model representing a blog post."""
+
+    title = models.CharField(max_length=100, null=True, unique=True)
+    content = models.TextField(null=True)
+    author = models.ForeignKey(
+        "tests.Author", related_name="posts", on_delete=models.CASCADE
+    )
+
+
+class Author(models.Model):
+    """Model representing a blog post Author."""
+    name = models.CharField(max_length=100)
+```
+
+If you want all author objects and their posts formatted as a dict, with
+title as key and content as value, you can run the following
+
+```python
+from json_agg import JSONObjectAgg
+
+from .models import Author
+
+
+queryset = Author.objects.annotate(
+    post_map=JSONObjectAgg("posts__title", "posts__content")
+).all()
+```
+
+In the example above, `Author` instances would have the attribute `post_map`, which would a
+a dict.
+
+A similar aggregator, `JSONArrayAgg`, is also available.
+
+Please see the [reference] for details.
+
+## Is this project for me?
+
+django-json-agg aims to improve the ergonomics for aggregating data as dicts or lists
+while still keeping other model instances. All that in a single query. One might argue
+it also adds some efficiency by building the dict/list in the database, instead of python side.
```

Comment:
I feel @infinitewarp might want to insert a "however..." here

Reply:
Well, since you mentioned it. 😆 

```suggestion
django-json-agg aims to improve the ergonomics for aggregating data as dicts or lists
while still keeping other model instances. All that in a single query. One might argue
it also adds some efficiency by building the dict/list in the database, instead of python side.

However, reliance on django-json-agg and JSON data structures in general for storing
"relational" data may be an anti-pattern or an indicator of a problematic or suboptimal
database schema design. Your time may be better spent refactoring your models to use
more native primitive data types and relations (foreign keys, indexes, etc).
```